### ISSUE TITLE
children! and BaseWidget Methods

### ISSUE THREAD

**Issue by Andi-K** [cybergeek94/kiss-ui#20]
If i try to use `Button::new().set_enabled(false)` or .set_name() i get the this error:

> <kiss_ui macros>:1:52: 1:62 error: the trait `core::convert::From<()>` is not implemented for the type > `kiss_ui::BaseWidget` [E0277]
> <kiss_ui macros>:1 ( $ ( $ child : expr ) , + , ) => ( [ $ ( $ child . into (  ) ) , + ] ) ; (  )
>                                                                       ^~~~~~~~~~
> <kiss_ui macros>:1:1: 2:14 note: in expansion of children!

I don't know how to try it without using children!
But i think is because Button don't  implement set_enabled().

**Comment by cybergeek94** [comment]
Ah yes, I realized late last night might be an issue. It's not exactly `.set_enabled()`'s fault, it's how it's called. See, `.set_enabled()` is implmented on `BaseWidget` and callable on the other widget types through their `DerefMut<Target=BaseWidget>` impls. It returns `&mut BaseWidget` for method chaining, but the result of each expression in `children![]` needs to be an owned widget type.

You can work around this by constructing the widget in a block:

```rust
children! [
    {
        let mut button = Button::new();
        button.set_enabled(false);
        button
    }
]
```

I've had to do that in a couple examples. It does work, but it's incredibly clunky. I have a fix in mind, but I'll need to do some refactoring. 

Basically, I'd move all the methods from `BaseWidget` into a new `Widget` trait, and implement that for all widget types. Then all those setter methods can take `self` by-value and return `Self` so that you can do method-chaining like this with methods from `Widget` *and* from the actual widget type.

It's another import, but I'm thinking about creating a `prelude` module that users can glob-import to get some of the more common types, such as `Widget` and `BaseWidget`.

**Comment by cybergeek94** [comment]
I've got the solution in #23.

**Comment by cybergeek94** [comment]
Okay, should be fixed now. Give it a try.

**Comment by Andi-K** [comment]
Hey that's great! :+1: 

Works perfect, thank you! :)
